meson.build

```diff
@@ -163,11 +163,41 @@ endif
 xsf = subproject('xsf')
 xsf_dep = xsf.get_variable('xsf_dep')
 
-boost_math_dep = dependency(
-  'boost',
-  version : '1.88.0',
-  fallback : ['boost_math', 'boost_math_dep'],
-)
+use_system_libraries = get_option('use-system-libraries')
+auto_system_libraries = false
+
+if use_system_libraries.contains('none')
+  use_system_libraries = ['none']
+elif use_system_libraries.contains('auto')
+  auto_system_libraries = true 
+endif
+
+if auto_system_libraries or use_system_libraries.contains('boost.math')
+  boost_math_dep = dependency(
+    'boost',
+    version : '1.88.0',
+    fallback : ['boost_math', 'boost_math_dep'],
+  )
+else
+  boost_math = subproject(
+    'boost_math',
+    version : '1.88.0',
+  )
+  boost_math_dep = boost_math.get_variable('boost_math_dep')
```

Comment:
These `dependency` and `subproject` calls may be easier to read if they're on a single line, especially if it becomes `if-elif-else`. 

If you prefer to leave it like this, also fine with me.

Reply:
Yep, LGTM now! resolving.